src/components/pages/ImageClassification/ImageClassificationTable/ImageClassificationContainer.js

```diff
@@ -4,23 +4,74 @@ import ImageUploadModal from '../ImageUploadModal/ImageUploadModal'
 import { ButtonPrimary } from '../../../generic/buttons'
 import { IconUpload } from '../../../icons'
 import { ButtonContainer, IconContainer } from './ImageClassificationObservationTable.styles'
+import { EXCLUDE_PARAMS_FOR_GET_ALL_IMAGES_IN_COLLECT_RECORD } from '../imageClassificationConstants'
+import { getIsImageProcessed } from '../getIsImageProcessed'
+import { useDatabaseSwitchboardInstance } from '../../../../App/mermaidData/databaseSwitchboard/DatabaseSwitchboardContext'
+import { useHttpResponseErrorHandler } from '../../../../App/HttpResponseErrorHandlerContext'
+import { useParams } from 'react-router-dom'
 
 const ImageClassificationContainer = (props) => {
+  const [images, setImages] = useState([])
   const [isModalOpen, setIsModalOpen] = useState(false)
-  const [uploadedFiles, setUploadedFiles] = useState([])
   const [isUploading, setIsUploading] = useState(false)
+  const [uploadedFiles, setUploadedFiles] = useState([])
+  const { databaseSwitchboardInstance } = useDatabaseSwitchboardInstance()
+  const { projectId, recordId } = useParams()
+  const handleHttpResponseError = useHttpResponseErrorHandler()
 
   const handleFilesUpload = (files) => {
     setUploadedFiles([...uploadedFiles, ...files])
     setIsModalOpen(false)
   }
 
+  const pollCollectRecordUntilAllImagesProcessed = () => {
+    if (!databaseSwitchboardInstance || !handleHttpResponseError || !projectId || !recordId) {
+      throw new Error('pollCollectRecordUntilAllImagesProcessed has missing dependencies')
+    }
+    let intervalId
+
+    const pollCollectRecordForImages = async () => {
+      try {
+        const response = await databaseSwitchboardInstance.getAllImagesInCollectRecord(
+          projectId,
+          recordId,
+          EXCLUDE_PARAMS_FOR_GET_ALL_IMAGES_IN_COLLECT_RECORD,
+        )
+
+        setImages(response.results)
+
+        const areAllImagesProcessed =
+          response.results.length &&
+          response.results.every((file) => getIsImageProcessed(file.classification_status?.status))
```

Comment:
This every passed if using an empty array! So we needed to also check for array length if we want to be able to use this function recursively (re poll the collect records). Crazy JS!

Reply:
(this is mostly copied code with some minor renaming from the other file, sans being triggered by an effect. Its now triggered by user upload directly)